src/test/java/ch/wisv/areafiftylan/EventRestIntegrationTest.java

```diff
@@ -0,0 +1,769 @@
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_BAD_REQUEST);
+        //@formatter:on
+
+        Assert.assertEquals(1, eventRepository.count());
+    }
+
+    @Test
+    public void testAddEventMissingTeamSizeAsAdmin() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_BAD_REQUEST);
+        //@formatter:on
+
+        Assert.assertEquals(1, eventRepository.count());
+    }
+
+    @Test
+    public void testAddEventMissingTeamLimit() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_BAD_REQUEST);
+        //@formatter:on
+
+        Assert.assertEquals(1, eventRepository.count());
+    }
+
+    @Test
+    public void testAddEventEmptyBody() {
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE);
+        //@formatter:on
+
+        Assert.assertEquals(1, eventRepository.count());
+    }
+
+    //endregion
+
+    //region Test Change Events
+
+    @Test
+    public void testChangeEventAsAnon() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "Changed Event");
+        eventDTO.put("teamSize", "3");
+        eventDTO.put("teamLimit", "11");
+
+        //@formatter:off
+        given().
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            put(EVENTS_ENDPOINT + "/" + testEvent.getId()).
+        then().
+            statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+
+        Event changedEvent = eventRepository.findOne(testEvent.getId());
+
+        Assert.assertEquals(testEvent.getEventName(), changedEvent.getEventName());
```

Comment:
teamSize is also changed. I know this would be a weird case, but I'd check the equality of this property as well.